src/Compiler/Service/IncrementalBuild.fs

```diff
@@ -358,26 +358,23 @@ type BoundModel private (tcConfig: TcConfig,
         let hasSig = this.BackingSignature.IsSome
 
         // If partial checking is enabled and we have a backing sig file, then use the partial state. The partial state contains the sig state.
-        if tcInfoNode.HasFull && enablePartialTypeChecking && hasSig then
-            let newTcInfoStateOpt =
-                match tcInfoNode with
-                | TcInfoNode(_, fullGraphNode) -> 
-                    let tcInfo, _ = fullGraphNode.TryPeekValue().Value
-                    Some(PartialState tcInfo)
-
-            BoundModel(
-                tcConfig,
-                tcGlobals,
-                tcImports,
-                keepAssemblyContents, keepAllBackgroundResolutions,
-                keepAllBackgroundSymbolUses,
-                enableBackgroundItemKeyStoreAndSemanticClassification,
-                enablePartialTypeChecking,
-                beforeFileChecked,
-                fileChecked,
-                prevTcInfo,
-                syntaxTreeOpt,
-                newTcInfoStateOpt)
+        if enablePartialTypeChecking && hasSig then
+            match tcInfoNode.TryGetFull() with
+            | ValueSome (tcInfo, _) when enablePartialTypeChecking && hasSig ->
```

Comment:
```suggestion
            | ValueSome (tcInfo, _) ->
```
If we keep the `if` then we can remove this `when` :)

Reply:
oh yeah :)